File: ly-server/router/framework/host.go
package framework

import (
	"ly-server/api"
	"ly-server/middlewares"

	"github.com/gin-gonic/gin"
)

type HostRouter struct{}

func (r *HostRouter) InitHostRouter(Router *gin.RouterGroup) {
	hostRouter := Router.Group("host").Use(middlewares.OperationRecord())
	hostRouterWithoutRecord := Router.Group("host")
	hostApi := api.ApiGroupApp.FrameworkApiGroup.HostApi
	{
		hostRouter.POST("register", hostApi.Register)        // 执行主机注册
		hostRouter.POST("status/modify", hostApi.StatusModify)        // 执行主机状态变更
		hostRouter.POST("checkHostPermission", hostApi.CheckHostPermission)   // 检查是否有host权限

	}
	{
		hostRouterWithoutRecord.POST("list", hostApi.GetHostList)
		hostRouterWithoutRecord.POST("getHostByHostID", hostApi.GetHostByHostID)   // 根据hostid获取host
		hostRouterWithoutRecord.POST("getHostByUserID", hostApi.GetHostByUserID)   // 根据用户id获取host
		hostRouterWithoutRecord.POST("getHostStatus", hostApi.GetHostStatus)   // 获取主机状态
	}
}

File: ly-web/src/api/host.ts
import service from "@/utils/request"

// @Summary 获取主机列表
export const getHostList = (data: any) => {
    return service({
        url: '/host/list',
        method: 'post',
        data: data
    })
}

export const checkHostPermission = (data: any) => {
    return service({
        url: '/host/checkHostPermission',
        method: 'post',
        data: data
    })
}

// @Summary 根据用户id获取主机
export const getHostByUserId = (data: any) => {
    return service({
        url: '/host/getHostByUserID',
        method: 'post',
        data: data
    })
}

// @Summary 根据host id获取主机
export const getHostByHostID = (data: any) => {
    return service({
        url: '/host/getHostByHostID',
        method: 'post',
        data: data
    })
}

// @Summary 执行机状态变更
export const hostStatusModify = (data: any) => {
    return service({
        url: '/host/status/modify',
        method: 'post',
        data: data
    })
}

// @Summary 获取执行机状态
export const getHostStatus = (data: any) => {
    return service({
        url: '/host/getHostStatus',
        method: 'post',
        data: data
    })
}